src/LLVM_intrinsics.jl

```diff
@@ -0,0 +1,647 @@
+    ret <$M x $(d[T])> %res
+    """
+    return :(
+        $(Expr(:meta, :inline));
+        Base.llvmcall($s, LVec{$M, T}, Tuple{LVec{N, T}, LVec{N, T}}, x, y)
+    )
+end
+
+@generated function shufflevector(x::LVec{N, T}, ::Val{I}) where {N, T, I}
+    shfl = _shuffle_vec(I)
+    M = length(I)
+    s = """
+    %res = shufflevector <$(N) x $(d[T])> %0, <$N x $(d[T])> undef, <$M x i32> <$shfl>
+    ret <$M x $(d[T])> %res
+    """
+    return :(
+        $(Expr(:meta, :inline));
+        Base.llvmcall($s, LVec{$M, T}, Tuple{LVec{N, T}}, x)
+    )
+end
+
+@generated function constantvector(v::T, y::Type{LVec{N, T}}) where {N, T}
+    s = """
+    %2 = insertelement <$N x $(d[T])> undef, $(d[T]) %0, i32 0
+    %res = shufflevector <$N x $(d[T])> %2, <$N x $(d[T])> undef, <$N x i32> zeroinitializer
+    ret <$N x $(d[T])> %res
+    """
+    return :(
+        $(Expr(:meta, :inline));
+        Base.llvmcall($s, LVec{N, T}, Tuple{T}, v)
+    )
+end
+
+#########################
+# Conversion Operations #
+#########################
+
+const CAST_SIZE_CHANGE_FLOAT = [
+    (:fptrunc, >)
+    (:fpext, <)
+]
+
+const CAST_SIZE_CHANGE_INT = [
+    (:trunc, >)
+    (:zext, <)
+    (:sext, <)
+]
+
+for (fs, c) in zip([CAST_SIZE_CHANGE_FLOAT, CAST_SIZE_CHANGE_INT],
+                   [FloatingTypes,          IntegerTypes])
+    for (f, criteria) in fs
+        @eval @generated function $f(::Type{LVec{N, T2}}, x::LVec{N, T1}) where {N, T1 <: $c, T2 <: $c}
+            sT1, sT2 = sizeof(T1) * 8, sizeof(T2) * 8
+            # Not changing size is not allowed
+            if !$criteria(sT1, sT2)
+                return :(error("size of conversion type ($T2: $sT2) must be $($criteria) than the element type ($T1: $sT1)"))
+            end
+            ff = $(QuoteNode(f))
+            s = """
+            %2 = $ff <$(N) x $(d[T1])> %0 to <$(N) x $(d[T2])>
+            ret <$(N) x $(d[T2])> %2
+            """
+            return :(
+                $(Expr(:meta, :inline));
+                Base.llvmcall($s, LVec{N, T2}, Tuple{LVec{N, T1}}, x)
+            )
+        end
+    end
+end
+
+const CONVERSION_FLOAT_TO_INT = [
+    :fptoui,
+    :fptosi
+]
+
+const CONVERSION_INT_TO_FLOAT = [
+    :uitofp,
+    :sitofp
+]
+
+for (fs, (from, to)) in zip([CONVERSION_FLOAT_TO_INT,       CONVERSION_INT_TO_FLOAT],
+                           [(FloatingTypes, IntegerTypes), (IntegerTypes, FloatingTypes)])
+    for f in fs
+        @eval @generated function $f(::Type{LVec{N, T2}}, x::LVec{N, T1}) where {N, T1 <: $from, T2 <: $to}
+            ff = $(QuoteNode(f))
+            s = """
+            %2 = $ff <$(N) x $(d[T1])> %0 to <$(N) x $(d[T2])>
+            ret <$(N) x $(d[T2])> %2
+            """
+            return :(
+                $(Expr(:meta, :inline));
+                Base.llvmcall($s, LVec{N, T2}, Tuple{LVec{N, T1}}, x)
+            )
+        end
+    end
+end
+
+
+###########
+# Bitcast #
+###########
```

Comment:
We also need a trunc-and-bitcast intruction to represent vpmovmskb.

That would take `<n x i8>`, truncate it to `<n x i1>`, zero-pad it, and cast it to e.g. `i32` (or avx2). This can afaik not be reproduced with the other intrinsics here.

Likewise, we probably want an operation that takes `<n x i8>`, truncates to `<n x i1>` and `sext`s to `<n x i8>`;  possibly extending even more. This is in order to reduce mismatch between julia/llvm and x86 semantics (x86 tends to set all the bits, i.e. `sext`; julia has no representation of `<n x i1>`, and julia's Bool uses `zext` -style `<n x i8>`).